packages/ui/src/hooks/useDetailsTabs.tsx

```diff
@@ -1,32 +1,44 @@
 import { useEffect, useState } from 'react';
 import { Status } from '@bull-board/api/typings/app';
 import { STATUSES } from '@bull-board/api/src/constants/statuses';
+import { useSettingsStore } from './useSettings';
 
-const regularItems = ['Data', 'Options', 'Logs'] as const;
+export const availableJobTabs = ['Data', 'Options', 'Logs', 'Error'] as const;
 
-export type TabsType = typeof regularItems[number] | 'Error';
+export type TabsType = (typeof availableJobTabs)[number];
+
+const FALLBACK_TAB: TabsType = 'Data';
 
 export function useDetailsTabs(currentStatus: Status, isJobFailed: boolean) {
   const [tabs, updateTabs] = useState<TabsType[]>([]);
-  const [selectedTabIdx, setSelectedTabIdx] = useState(0);
-  const selectedTab = tabs[selectedTabIdx];
+  const { defaultJobTab } = useSettingsStore();
+
+  const [selectedTab, setSelectedTab] = useState<TabsType>(defaultJobTab);
```

Comment:
defaultJobTab can be `Error` but the job is not failed, it should ignore the settings default if there is no available tab.

```js
const [selectedTab, setSelectedTab] = useState<TabsType>(tabs.find(tab => tab === defaultJobTab) || tabs[0]);
```